Jacky's Pi Sys/pi-systems_sensor-reader.py

```diff
@@ -68,47 +71,59 @@ def __update_sensor_data(self):
+                    self.humidity_readings.append([sensor_data[5], time.time()])
                 if validData & 1 << 5:
-                    temperatureVal += sensor_data[6]
-                    numReadingsTemp += 1
+                    self.temp_readings.append([sensor_data[6], time.time()])
                 if validData & 1 << 4:
-                    pressureVal += sensor_data[7]
-                    numReadingsPressure += 1
+                    self.pressure_readings.append([sensor_data[7], time.time()])
                 if validData & 1 << 3:
-                    C02Val += sensor_data[8]
-                    numReadingsC02 += 1
+                    self.C02_readings.append([sensor_data[8], time.time()])
             except ValueError as ve:
                 print("Invalid object read from I2C.\n\tStackTrace: " +
                       str(ve) + "\n\tSkipping line...")
 
-        # computes average of all readings if they have been read at least once
-        if numReadings02 != 0:
-            O2Val = O2Val / numReadingsO2
-        if numReadingsHumidity != 0:
-            humidityVal = humidityVal / numReadingsHumidity
-        if numReadingsTemp != 0:
-            temperatureVal = temperatureVal / numReadingsTemp
-        if numReadingsPressure != 0:
-            pressureVal = pressureVal / numReadingsPressure
-        if numReadingsC02 != 0:
-            C02Val = C02Val / numReadingsC02
-
-        # stores average readings into dictionary
+        # loop through each readings list and check last element with current
+        # element time, and if diff greater than keep_track_seconds, remove
+        # current element list from readings list and then sums value
+        # of all readings after trim
+        O2_val = self.trim_sum_list(self.O2_readings)
+        humidity_val = self.trim_sum_list(self.humidity_readings)
+        temperature_val = self.trim_sum_list(self.temp_readings)
+        pressure_val = self.trim_sum_list(self.pressure_readings)
+        C02_val = self.trim_sum_list(self.C02_readings)
+
+        # computers average of all readings after sum computed
+        # and stores average readings into dictionary
         self.sensor_data = {
-            'O2': O2Val,
-            'humidity': humidityVal,
-            'temperature': temperatureVal,
-            'pressure': pressureVal,
-            'CO2': C02Val
+            'O2': O2_val / len(self.O2_readings),
+            'humidity': humidity_val / len(self.humidity_readings),
+            'temperature': temperature_val / len(self.temp_readings),
+            'pressure': pressure_val / len(self.pressure_readings),
+            'CO2': C02_val / len(self.C02_readings)
         }
 
-    def error_check(self):
+    # function to check amount of time we want to keep history of readings
+    # and remove sublists in list that have expired in terms of time to
+    # keep data and then sums value of valid readings and returns
+    def trim_sum_list(self, list):
```

Comment:
I think you're incorrectly referring to list elements here. You're using _index(i)_, which attempts to find the value _i_ in the list and returns its index. I think you want to access the element at index _i_, which is done using _list[i]_

**As it stands, list.index(-1) will never work, because you are looking for an integer in a list of lists.